### PR TITLE
feature(wc_yotpo.php): created possibility of v3 widgets usage

### PR DESCRIPTION
**Pull request by p-bartosz**

### CODE REVIEW

File: trunk/wc_yotpo.php [hunk 1]
```diff
@@ -103,30 +104,80 @@ function wc_yotpo_uninstall() {
...
-                    data-currency='".get_woocommerce_currency()."'></div>";
-		echo $yotpo_div;
+	if($product->get_reviews_allowed() == true) {
+		echo generate_reviews_widget_code();
+	}						
+}
+function generate_reviews_widget_code() {
+	$yotpo_settings = get_option('yotpo_settings',wc_yotpo_get_degault_settings());
+	$use_v3_widgets = $yotpo_settings['use_v3_widgets'];
+	$reviews_widget_id = $yotpo_settings['reviews_widget_id'];
```
Comment (LiorGingi) [inline trunk/wc_yotpo.php]: Why do you get the widget ID here? should be part of the `generate_v3_reviews_widget_code` method.

File: trunk/wc_yotpo.php [hunk 2]
```diff
@@ -103,30 +104,80 @@ function wc_yotpo_uninstall() {
...
+		data-yotpo-currency='".get_woocommerce_currency()."'
+		data-yotpo-description='".$product_data['description']."'
+	></div>";
+}
+// Q&A WIDGET
+function wc_yotpo_show_qna_widget() {		 
+	global $product;
+	$yotpo_settings = get_option('yotpo_settings',wc_yotpo_get_degault_settings());
+	$use_v3_widgets = $yotpo_settings['use_v3_widgets'];
+	$qna_widget_id = $yotpo_settings['qna_widget_id'];
```
Comment (LiorGingi) [inline trunk/wc_yotpo.php]: Getting the widget ID should be part of the `generate_v3_qna_widget_code` method.

File: trunk/wc_yotpo.php [hunk 3]
```diff
@@ -139,32 +190,61 @@ function wc_yotpo_load_js(){
 	}
 }
 function wc_yotpo_show_qa_bottomline() {
+	$use_v3_widgets = get_option('yotpo_settings',wc_yotpo_get_degault_settings())['use_v3_widgets'];
+	if ($use_v3_widgets) {
+		return;
+	}
```
Comment (LiorGingi) [inline trunk/wc_yotpo.php]: Why return? don't we want to show the Q&A bottomline for V3 customers?

File: trunk/wc_yotpo.php [hunk 4]
```diff
@@ -139,32 +190,61 @@ function wc_yotpo_load_js(){
...
-	   				data-lang='".$product_data['lang']."'></div>";
+		$yotpo_div = generate_star_ratings_widget_code();
 		echo $yotpo_div;	
 	}	
 				
 }
+function generate_star_ratings_widget_code() {
+	$yotpo_settings = get_option('yotpo_settings',wc_yotpo_get_degault_settings());
+	$use_v3_widgets = $yotpo_settings['use_v3_widgets'];
+	$reviews_widget_id = $yotpo_settings['reviews_widget_id'];
```
Comment (LiorGingi) [inline trunk/wc_yotpo.php]: Star rating getting reviews widget ID?

File: trunk/wc_yotpo.php [hunk 5]
```diff
@@ -385,9 +465,13 @@ function wc_yotpo_conversion_track($order_id) {
 function wc_yotpo_get_degault_settings() {
     return array('app_key' => '',
         'secret' => '',
+				'reviews_widget_id' => '',
+				'qna_widget_id' => '',
+				'ratings_widget_id' => '',
```
Comment (LiorGingi) [inline trunk/wc_yotpo.php]: The indentation seems wrong here.

File: trunk/templates/wc-yotpo-settings.php [hunk 6]
```diff
@@ -132,7 +135,7 @@ function wc_display_yotpo_settings($success_type = false) {
                      </tr>
                      <tr valign='top' id='yotpo-sync-widget-ids-row' style='display:none'>
                        <th scope='row'><div>Synchronise the widgets' codes:</div></th>
```
Comment (LiorGingi) [inline trunk/templates/wc-yotpo-settings.php:182]: Did this wording go through your PM?

File: trunk/templates/wc-yotpo-settings.php [hunk 7]
```diff
@@ -135,7 +138,12 @@ function wc_display_yotpo_settings($success_type = false) {
...
+                       <!-- <td><button type='button' id='yotpo-sync-widget-ids' class='button-secondary'>Sync</button></td> -->
                        <td><input type='submit' name='yotpo_sync_ids' value='Sync' class='button-primary' id='yotpo_sync_ids'/></td>
+                     </tr>
+                     <tr valign='top'>
+                       <th scope='row'><div>Reset the widgets' codes:</div></th>
+                       <td><input type='submit' name='yotpo_remove_ids' value='Reset' class='button-secondary' id='yotpo_remove_ids'/></td>
```
Comment (LiorGingi) [inline trunk/templates/wc-yotpo-settings.php]: Why should they remove it? Checking out the checkout should revert them to V2, shouldn't it?

File: trunk/wc_yotpo.php [hunk 8]
```diff
@@ -216,9 +216,12 @@ function generate_star_ratings_widget_code() {
 	$use_v3_widgets = $yotpo_settings['use_v3_widgets'];
 	$reviews_widget_id = $yotpo_settings['reviews_widget_id'];
 	$star_ratings_widget_id = $yotpo_settings['star_ratings_widget_id'];
-	return $use_v3_widgets && $reviews_widget_id && $star_ratings_widget_id
-		? generate_v3_star_ratings_widget_code($star_ratings_widget_id)
-		: generate_v2_star_ratings_widget_code();
+	if ($use_v3_widgets && $star_ratings_widget_id && $reviews_widget_id) {
+		return generate_v3_star_ratings_widget_code($star_ratings_widget_id);
+	}
+	if (!$use_v3_widgets) {
```
Comment (LiorGingi) [inline trunk/wc_yotpo.php]: Are there any further cases where nothing will be returned over here?
If not, this if statement is redundant.

**Review by LiorGingi — CHANGES_REQUESTED** [review]

**Comment by p-bartosz** [comment]
@LiorGingi left one comment
all the rest is outdated

**Review by LiorGingi — APPROVED** [review]

**Pull request merged**